src/wp-includes/block-supports/layout.php

```diff
@@ -41,45 +42,145 @@ function wp_register_layout_support( $block_type ) {
+				array_push(
+					$layout_styles,
+					array(
+						'selector'     => "$selector > *",
+						'declarations' => array(
+							'margin-block-start' => '0',
+							'margin-block-end'   => '0',
+						),
+					),
+					array(
+						'selector'     => "$selector$selector > * + *",
+						'declarations' => array(
+							'margin-block-start' => $gap_value,
+							'margin-block-end'   => '0',
+						),
+					)
+				);
+			}
+		}
+	} elseif ( 'constrained' === $layout_type ) {
+		$content_size    = isset( $layout['contentSize'] ) ? $layout['contentSize'] : '';
+		$wide_size       = isset( $layout['wideSize'] ) ? $layout['wideSize'] : '';
+		$justify_content = isset( $layout['justifyContent'] ) ? $layout['justifyContent'] : 'center';
 
 		$all_max_width_value  = $content_size ? $content_size : $wide_size;
 		$wide_max_width_value = $wide_size ? $wide_size : $content_size;
 
 		// Make sure there is a single CSS rule, and all tags are stripped for security.
-		$all_max_width_value  = safecss_filter_attr( explode( ';', $all_max_width_value )[0] );
-		$wide_max_width_value = safecss_filter_attr( explode( ';', $wide_max_width_value )[0] );
+		// TODO: Use `safecss_filter_attr` instead - once https://core.trac.wordpress.org/ticket/46197 is patched.
+		$all_max_width_value  = wp_strip_all_tags( explode( ';', $all_max_width_value )[0] );
+		$wide_max_width_value = wp_strip_all_tags( explode( ';', $wide_max_width_value )[0] );
+
+		$margin_left  = 'left' === $justify_content ? '0 !important' : 'auto !important';
+		$margin_right = 'right' === $justify_content ? '0 !important' : 'auto !important';
 
 		if ( $content_size || $wide_size ) {
-			$style  = "$selector > :where(:not(.alignleft):not(.alignright)) {";
-			$style .= 'max-width: ' . esc_html( $all_max_width_value ) . ';';
-			$style .= 'margin-left: auto !important;';
-			$style .= 'margin-right: auto !important;';
-			$style .= '}';
-
-			$style .= "$selector > .alignwide { max-width: " . esc_html( $wide_max_width_value ) . ';}';
-			$style .= "$selector .alignfull { max-width: none; }";
+			array_push(
+				$layout_styles,
+				array(
+					'selector'     => "$selector > :where(:not(.alignleft):not(.alignright):not(.alignfull))",
+					'declarations' => array(
+						'max-width'    => $all_max_width_value,
+						'margin-left'  => $margin_left,
+						'margin-right' => $margin_right,
+					),
+				),
+				array(
+					'selector'     => "$selector > .alignwide",
+					'declarations' => array( 'max-width' => $wide_max_width_value ),
+				),
+				array(
+					'selector'     => "$selector .alignfull",
+					'declarations' => array( 'max-width' => 'none' ),
+				)
+			);
+
+			if ( isset( $block_spacing ) ) {
+				$block_spacing_values = wp_style_engine_get_styles(
+					array(
+						'spacing' => $block_spacing,
+					)
+				);
+
+				// Handle negative margins for alignfull children of blocks with custom padding set.
+				// They're added separately because padding might only be set on one side.
```

Comment:
[See here](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#5-2-multi-line-comments) for the multi-line comment format.

In short:
```php
/*
 * First line.
 *
 * Last line.
 */ 
```

🔢 Applies to other instances in the PR.

Reply:
Thanks! I've updated each of the instances of multi-line comments.